nix/install.in

```diff
@@ -5,19 +5,22 @@
 # (which in turn creates and populates /nix).
 
 { # Prevent execution if this script was only partially downloaded
-
-unpack=nix-binary-tarball-unpack
+oops() {
+    echo "$0: $@" >&2
+    exit 1
+}
+unpack="$(mktemp -d nix-binary-tarball-unpack.XXXXXXXXXX || \
+          oops "Can't create tempdir for unpacking")"
+cleanup() {
+    rm -rf "$unpack"
+}
+trap cleanup EXIT
```

Comment:
You're right. As far as I can tell from the bash docs and experimentation, trap EXIT handles signal-caused exit of the script in bash correctly (in the example you link above exit on INT etc. is explicitly prevented by overriding with a no-op). Unfortunately it's underspecified by posix and other popular shells like dash and zsh don't behave this way (and the install instructions on the site pipe into sh not bash). 

Sigh. As far as I can tell that means there is no robust way to do cleanup on exit then (I believe even with your addition e.g. USR1 etc. will cause exit without cleanup and the signals themselves are not all portable). Pragmatically EXIT INT TERM QUIT should cover all "natural" cases, so I'll amend as you suggest.